gax-httpjson/src/test/java/com/google/api/gax/httpjson/ProtoRestSerializerTest.java

```diff
@@ -181,6 +195,75 @@ public void putQueryParam() {
     Truth.assertThat(fields).isEqualTo(expectedFields);
   }
 
+  @Test
+  public void putQueryParamComplexObject() {
+    Map<String, List<String>> fields = new HashMap<>();
+    Any fieldMask = Any.pack(FieldMask.newBuilder().addPaths("a.b.c").addPaths("d.e.f").build());
+    Any duration1 = Any.pack(Duration.newBuilder().setSeconds(1).setNanos(1).build());
+    Any duration2 = Any.pack(Duration.newBuilder().setSeconds(2).setNanos(2).build());
+    Field value =
+        Field.newBuilder()
+            .setNumber(2)
+            .setName("well_known_container")
+            .addOptions(Option.newBuilder().setName("duration").setValue(duration1).build())
+            .addOptions(Option.newBuilder().setName("duration").setValue(duration2).build())
+            .addOptions(Option.newBuilder().setName("mask").setValue(fieldMask).build())
+            .setCardinality(Cardinality.CARDINALITY_OPTIONAL)
+            .build();
+    requestSerializer.putQueryParam(fields, "object", value);
+
+    Map<String, List<String>> expectedFields = new HashMap<>();
+    expectedFields.put("object.name", Arrays.asList("well_known_container"));
+    expectedFields.put("object.number", Arrays.asList("2"));
+    expectedFields.put("object.options.name", Arrays.asList("duration", "duration", "mask"));
+    expectedFields.put(
+        "object.options.value.value", Arrays.asList("1.000000001s", "2.000000002s", "a.b.c,d.e.f"));
+    // used by JSON parser to obtain descriptors from this type url
+    expectedFields.put(
```

Comment:
This is an interesting edge case, I don't think we should include these type info in the query params, we probably don't want to configure an Any field as query params as well. Can you do a little more investigation on this?